src/main/java/org/apache/sling/xss/impl/AntiSamyPolicyAdapter.java

```diff
@@ -222,43 +224,28 @@ public boolean apply(String s) {
     private static Predicate<String> matchesPatternsOrLiterals(List<Pattern> patternList, boolean ignoreCase, List<String> literalList) {
         return new Predicate<String>() {
             public boolean apply(String s) {
-                // check if the string matches to the pattern
-                for (Pattern pattern : patternList) {
-                    if (pattern.matcher(s).matches()) {
-                        return true;
-                    }
-                }
-                // if the pattern does not match it goes through the literals
-                for (String string : literalList) {
-                    s = ignoreCase
-                        ? s.toLowerCase()
-                        : s;
-                    if (string.equals(s)) {
-                        return true;
-                    }
-                }
-                // if it neither matches the patterns nor the literals it returns false
-                return false;
+                // check if the string matches to the pattern or one of the literal
+                s = ignoreCase ? s.toLowerCase() : s;
```

Comment:
`String.toLowerCase` without an explicit locale can be dangerous, see https://stackoverflow.com/questions/11063102/using-locales-with-javas-tolowercase-and-touppercase for some discussion. Please figure out whether we can run in the default locale or need to use something like english.